src/main/java/com/lafaspot/ja3_4java/JA3Signature.java

```diff
@@ -0,0 +1,292 @@
+package com.lafaspot.ja3_4java;
+
+import java.nio.BufferUnderflowException;
+import java.nio.ByteBuffer;
+
+/**
+ * Generates JA3 signature based on the implementation described at https://github.com/salesforce/ja3.
+ *
+ */
+public final class JA3Signature {
+    /**
+     * Handshake byte.
+     */
+    private static final byte HANDSHAKE = 22;
+    /**
+     * Client hello.
+     */
+    private static final byte CLIENT_HELLO = 1;
+    /**
+     * Client hello random lenght.
+     */
+    private static final byte CLIENT_HELLO_RANDOM_LEN = 32;
+
+    /**
+     * Values to account for GREASE (Generate Random Extensions And Sustain Extensibility) as described here:
+     * https://tools.ietf.org/html/draft-davidben-tls-grease-01.
+     */
+    private static final int[] GREASE = new int[] { 0x0a0a, 0x1a1a, 0x2a2a, 0x3a3a, 0x4a4a, 0x5a5a, 0x6a6a, 0x7a7a, 0x8a8a, 0x9a9a, 0xaaaa, 0xbaba,
+            0xcaca, 0xdada, 0xeaea, 0xfafa };
+
+
+    /**
+     * Private constructor.
+     */
+    private JA3Signature() {
```

Comment:
Is this dependent on the TLS version? Will you support any version or only 1.2 and above??